web/src/pages/onboarding.tsx

```diff
@@ -216,23 +216,25 @@ const OnboardingPage = (props: Props): ReactElement => {
     } else {
       setPage({ current: 3, previous: 2 });
     }
+    await updateQueue?.queueProfileData(profileData);
```

Comment:
For these, I'd recommend using a local variable for both the setProfileData and the profileData that gets queued here.

SetState functions aren't actually synchronous so even though we're calling setProfileData up there, you can't guarantee that it will be set to the new value at this point in the code.

Reply:
Fair point. I'll make that update.